R/Plot_synergy.R

```diff
@@ -1,328 +1,332 @@
+               round(mean(scores.dose[.SliceAtLeastFirst(row.range),
+                                      .SliceAtLeastFirst(col.range)],
+                          na.rm = TRUE),
+                     3),
+               " (",data.method,")",
+               sep = ""))
+}
+
+# Convert 'val' to a slice.  However, guarantee that the slice starts at
+# 2 or higher.  So, both for val=c(1,5) and val=c(2,5), the function should
+# return 2:5.  The first two elements of list 'val' are taken to be the slice
+# endpoints; any other elements are ignored.  If 'val' is NULL, then the
+# returned slice is '-1' -- 'remove the first element'.
+.SliceAtLeastFirst <- function(val) {
+  return(if(is.null(val)) -1 else max(val[1],2):val[2])
+}
+
+# Returns a slice of 'val' specified by 'range', as numeric.
+# If 'range' is null, 'val' is wholly returned.  The first two elements of
+# 'range' are taken to be slice endpoints; any other values are ignored.
+# The desired portion of 'val' is converted to numeric form, then returned.  An
+# error occurs if any element of the selected slice of 'val' cannot be converted
+# to a number.
+.NumericApplyRange <- function(val, range) {
+  return(as.numeric(if(is.null(range)) val else val[range[1]:range[2]]))
+}
+
+# Returns a 2D slice of 'data' specified by 'row.range' and 'col.range'.  If
+# either range is NULL, the data for the corresponding axis of 'data' is
+# returned in full.  The row.range and col.range inputs are each treated as
+# lists whose first two elements are range endpoints, and whose remaining
+# elements (if any) are ignored.
+.ApplyRanges <- function(data, row.range, col.range) {
+  # Default value is 'remove the element just after the end of the list',
+  # for each dimension of the array (a no-op that permits cleaner syntax).
+  return(data[.RangeWithDefault(row.range, default=-(nrow(data)+1)),
+              .RangeWithDefault(col.range, default=-(ncol(data)+1))])
+}
+
+# If 'range' is NULL, return 'default', otherwise return a slice.
+# The endpoints of the slice are determined by assuming 'range' is a list, and
+# picking its first two elements.  Any additional elements are ignored.
+.RangeWithDefault <- function(range, default) {
+  return(if(is.null(range)) default else range[1]:range[2])
+}
+
+
+# TODO/REVIEW: The original logic checked into GitHub at the time this function
+# was written used flipped argument order for data.coord and krig.coord.
+# Furthermore, that logic appeared to transpose its output.  Please see the
+# corresponding unit test in test_that("extended_scores_works", {...}).  The
+# original logic swapped the locations of corner values '2' and '3'.  Looking
+# at the dose/response matrices, it seems that this transposition may have been
+# a bug.  With the code below, the contour and wireframe plots are similarly
+# transposed from their current appearance in the documentation, but seem to
+# gibe better with the documented dose/response matrices.
+
```

Comment:
the transpose is indeed a bug. The transpose was something from previous code which I should clean it up! Thanks for pointing this out!

Reply:
I'm not sure I understand.  Please see the comment I opened for the unit test for .ExtendedScores(), so I'm sure we're talking about the same thing.

Also, if you can determine expected unit test input/output data for a non-square case, it would be super-helpful.  I wasn't able to get that to work.